### DISCUSSION TITLE
(Everything) Add support/flag for using Alpha releases

### DISCUSSION THREAD

**Discussion by fabiorzfreitas** [chocolatey-community/chocolatey-packages#2289] [Ideas]
### Before starting, ensure that you have done the following.

- [X] I have verified that there is no other issue or discussion covering the feature or enhancement
- [X] I have verified there is no open pull request for the feature or enhancement
- [X] I have verified the package to improve is located in this repository

### What do you want to add?

Everything 1.4.1 was released in 2017 and had major bug fixes/improvements up to 2021, but as of today, it only gets minor fixes once every few months.

Meanwhile, 1.5.0.----alpha had its initial release in 2021, and it's being [actively developed in the forums](https://www.voidtools.com/forum/viewtopic.php?f=12&t=9787).

Devs say that v1.5.0.a won't interfere with a current 1.4.1 install, they should be side by side, but since it has been developed for over 2 years now and that there are some very interesting new features, I would very much like to ditch v1.4.1 and try the alpha: it's got 3 new releases already this month, they're always posted to the form post above.

Maybe this is would be preferable to be implemented via an `everything-alpha` package, but a simple `alpha` flag for these releases on this package would be easier.

Thanks for your consideration!

### What problem will the feature/enhancement solve?

Again, the [forums link](https://www.voidtools.com/forum/viewtopic.php?f=12&t=9787#improved) has a nice list of improvements.

### If the feature or enhancement is approved:

- [ ] I am willing to create a pull request to implement the feature or enhancement

### Additional Context

I'm not sure if I'd be willing to create a pull request. I don't really know much about Chocolatey, so it would depend on how much studying I'd have to do before being able to create a decent PR. If it's feasible within my current schedule, sure, but otherwise, I might have to installing v.1.5.0a outside Chocolatey :/

**Comment by pauby** [comment]
The `everything` package is only publishing stable releases, and stable releases are the only thing they are publishing on their website. They are posting the development releases on their forums, which isn't a great experience for automatically updating packages. Once they transition 1.5 to a stable release, the package update scripts will pick it up.

_In my opinion_ there isn't anything that we should do for the Community Packages repository.

There isn't anything to stop you creating an `everything-development` package (don't use `alpha` or `beta` in the package name), however.

**Reply by fabiorzfreitas** [reply]
After a little more learning about Chocolatey, I'm returning to suggest the implementation of a `--prerelease` switch for installing 1.5 alpha releases.

I also intend to go ask in their forums for a static link (or at least something outside the forums), so this idea becomes more feasible!

---

PS: I also just noticed they use a very simple versioning scheme: the releases are always 1.5.0.xxxx, and the following is 1.5.0.xxxx+1, it has been like this since the initial release. Also, the release file themselves are not stored on the forums, but on the main website. 1.5.1 or moving to stable are issues to consider, but this would be just a `--pre`, after all...

---

Update: they actually provided [static links for the latest releases](https://www.voidtools.com/forum/viewtopic.php?t=14152) on the main website, outside of the forum (not accessible by browsing the website, but available with the link itself).

Ideally, I'd still like to see it added to the main package with a `--prerelease` switch, but if it's still out of the scope of this package, I will try to create a separate package for 1.5.
